### PR TITLE
Document the need for migrating tool to user lockfiles

### PR DESCRIPTION
**Pull request by lilatomic**
We did some refactoring and the old steps no longer worked. It's come up a few times. We can be helpful and provide the info everywhere. Also we didn't update the steps for setting up IDEs.

### CODE REVIEW

**Comment by lilatomic** [comment]
I'm not sure if the updates to the current docs shoudl be done here or in the main repo or in the help-text of the Export goal

**Comment by thejcannon** [comment]
For live branches, you'd update the pants repo. For no longer active branches you'd update this repo. (There's ongoing chatter on his to make this easier)

**Comment by benjyw** [comment]
Also, I agree that it would be better if we found a way to allow export of the built-in non-resolve lockfiles.

File: docs/docs/using-pants/setting-up-an-ide.mdx [hunk 1]
```diff
@@ -49,7 +49,7 @@ The `--py-resolve-format=symlinked_immutable_virtualenv` option symlinks to an i
 
 ### Tool virtualenvs
 
-`pants export` can also create a virtualenv for each of the Python tools you use via Pants, such as `black`, `isort`, `pytest`, `mypy`, `flake8` and so on (you can run `/pants help tools` to get a list of the tools Pants uses). Use the tool name as the resolve name argument to the `--resolve` flag. This allows you to configure your editor to use the same version of the tool as Pants does for workflows like formatting on save.
+`pants export` can also create a virtualenv for each of the Python tools you use via Pants, such as `black`, `isort`, `pytest`, `mypy`, `flake8` and so on. This allows you to configure your editor to use the same version of the tool as Pants does for workflows like formatting on save. Follow the instructions for creating a tool lockfile  [here](../../python/overview/lockfiles#lockfiles-for-tools).
```
Comment (huonw) [inline docs/docs/using-pants/setting-up-an-ide.mdx:52]: It's a bit of nitpick, but what do you think about advice like https://usability.yale.edu/web-accessibility/articles/links#link-text which might suggest a version like the following might be better?
```suggestion
`pants export` can also create a virtualenv for each of the Python tools you use via Pants, such as `black`, `isort`, `pytest`, `mypy`, `flake8` and so on. This allows you to configure your editor to use the same version of the tool as Pants does for workflows like formatting on save. Follow [the instructions for creating a tool lockfile](../../python/overview/lockfiles#lockfiles-for-tools).
```

(And same in the other places below.)

**Review by huonw — APPROVED** [review]
Thanks for this!

**Comment by huonw** [comment]
(wait oops, wrong PR 😅 that was meant to be on https://github.com/pantsbuild/pants/pull/20604)

**Comment by huonw** [comment]
This was merged over in https://github.com/pantsbuild/pants/pull/20604 and cherry picked to 2.18.x, 2.19.x and 2.20.x, and will thus appear on the doc site on the next releases from those branches (#148 is related).

Thanks @lilatomic !

**Pull request closed without merging**